src/algorithms.rs

```diff
@@ -655,9 +692,9 @@ fn div_rem_core(mut a: BigUint, b: &BigUint) -> (BigUint, BigUint) {
         let mut prod = b * &q0;
 
         while cmp_slice(&prod.data[..], &a.data[j..]) == Greater {
-            let one: BigUint = One::one();
-            q0 = q0 - one;
-            prod = prod - b;
+            let one: BigDigit = 1;
```

Comment:
Does this library draw a distinction between `rust`'s `1` literal and the ring identity element? I think I remember @cuviper calling the later "logical one" at some point. Do we think it is worth implementing `num_traits::identities::One` on `BigDigit` just to make the point that these two ones are different? It might allow more freedom in the future, if we were to change the underlying type of `BigDigit`.

Reply:
Actually, I'm not entirely sure what BigDigit is in all cases. I think it is an alias for u32 or u64 depending on platform, but both of those already implement `One`.

Reply:
`BigDigit` is an internal-only detail at this point, just `u32` or `u64`. Since it's a `type` alias, it behaves exactly as the primitive types, including that it accepts literals like `1`. So at this time, I think going through `One` is needless abstraction.

> It might allow more freedom in the future, if we were to change the underlying type of `BigDigit`.

I don't expect we'd ever change to something that's not a primitive integer, but if we do, we'll get a nice compiler error during the refactoring. :) (i.e. this wouldn't fail silently)